collect_app/src/test/java/org/odk/collect/android/permissions/PermissionDeniedDialogTest.kt

```diff
@@ -0,0 +1,94 @@
+
+            // Button neutral
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEUTRAL)).visibility, `is`(View.VISIBLE))
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEUTRAL)).text, `is`(ApplicationProvider.getApplicationContext<Application>().getString(R.string.open_settings)))
+
+            // Button negative
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEGATIVE)).visibility, `is`(View.GONE))
+
+            // Title
+            val titleId: Int = ApplicationProvider.getApplicationContext<Application>().resources.getIdentifier("alertTitle", "id", ApplicationProvider.getApplicationContext<Application>().packageName)
+            assertThat((it.dialog!!.findViewById(titleId) as TextView).text, `is`(ApplicationProvider.getApplicationContext<Application>().getString(R.string.camera_runtime_permission_denied_title)))
+
+            // Message
+            assertThat((it.dialog!!.findViewById(android.R.id.message) as TextView).text, `is`(ApplicationProvider.getApplicationContext<Application>().getString(R.string.camera_runtime_permission_denied_desc)))
+        }
+    }
+
+    @Test
+    fun `The dialog should be dismissed after clicking on a device back button`() {
+        val scenario = DialogFragmentTest.launchDialogFragment(PermissionDeniedDialog::class.java, args)
+        scenario.onFragment {
+            assertThat(it.dialog!!.isShowing, `is`(true))
+            onView(isRoot()).perform(pressBack())
+            assertThat(it.dialog, `is`(nullValue()))
+        }
+    }
+
+    @Test
+    fun `The dialog should be dismissed after clicking on the POSITIVE button`() {
+        val scenario = DialogFragmentTest.launchDialogFragment(PermissionDeniedDialog::class.java, args)
+        scenario.onFragment {
+            assertThat(it.dialog!!.isShowing, `is`(true))
+            (it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_POSITIVE)).performClick()
+            RobolectricHelpers.runLooper()
+            assertThat(it.dialog, `is`(nullValue()))
+        }
+    }
+
+    @Test
+    fun `System settings should be opened after clicking on the NEUTRAL button`() {
+        val scenario = DialogFragmentTest.launchDialogFragment(PermissionDeniedDialog::class.java, args)
+        scenario.onFragment {
+            Intents.init()
+
+            assertThat(it.dialog!!.isShowing, `is`(true))
+            (it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEUTRAL)).performClick()
+
+            Intents.intended(IntentMatchers.hasAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS))
+            Intents.intended(IntentMatchers.hasData(Uri.fromParts("package", ApplicationProvider.getApplicationContext<Application>().packageName, null)))
+
+            Intents.release()
```

Comment:
We can use `RecordedIntentsRule` here I think.

Reply:
I had to move `RecordedIntentsRule` to `testshared` but ok now it is possible.